python/cudf/cudf/tests/test_index.py

```diff
@@ -190,6 +190,40 @@ def test_index_rename():
     assert_eq(pds, gds)
 
 
+def test_index_rename_inplace():
+    pds = pd.Index([1, 2, 3], name="asdf")
+    gds = as_index(pds)
+
+    # inplace=False should yield a deep copy
+    gds_renamed_deep = gds.rename("new_name", inplace=False)
+    gds._values.data.mem = GenericIndex([2, 3, 4])._values.data.mem
+
+    assert (gds_renamed_deep.values == [1, 2, 3]).all()
+
+    # inplace=True should yield a shallow copy
+    gds_renamed_shallow = gds
+    gds.rename("new_name", inplace=True)
+    gds._values.data.mem = GenericIndex([3, 4, 5])._values.data.mem
+
+    assert (gds_renamed_shallow.values == [3, 4, 5]).all()
```

Comment:
Would suggest updating comment and variable name as `inplace=True` doesn't yield a shallow copy.